### DISCUSSION TITLE
CP: Custom Source: Filter by multiple categories / related entry ID's

### DISCUSSION THREAD

**Discussion by adrianjean** [craftcms/cms#13993] [Q&A] [answered]
In the `CP- > Entries` I created a custom source. 

I added Entry Type as a filter, which shows entries for `Article`.

I added a filter for `Article Topic`, which is an entry reference field on `Article`.

I can tag multiple `Article Topic`, for each `Article`. So in my Custom Source, I want to only show `Article` that are tagged with 3 specific `Article Topic`. I know their ID's as: `34, 35, 36`

In my Custom Source filter, if I put the single id of `34` — I get a listing of all the `Article` related to Article Topic `34`. Good!

If I add another ID to the filter, I get not results. I have tried the following formats as input to the filter:

```
34,35,36
['34','35','36']
```

None show any results. Is it possible to create a custom source that only shows me entires related to a specific list of multiple related ID's?

<img width="423" alt="Screenshot 2023-12-05 at 7 09 30 PM" src="https://github.com/craftcms/cms/assets/3161182/973b0a37-199b-4b39-b981-f6e7e51db939">

**Comment by BenParizek** [comment]
You might want to post what you are doing in your ConditionRule modifyQuery or matchElement methods.

Multiple ids can require a different syntax than a single ID and since your filter seems to work with a single ID it suggests that you might need to update that.

A generic where statement for multiple IDs might look like this:

```->where(['in', 'id', $ids])```

You can find more examples in the Yii docs:
https://www.yiiframework.com/doc/guide/2.0/en/db-query-builder#where

Or by searching for something like `where(['in'` in the Craft codebase.

**Comment by brandonkelly** [comment, marked as answer]
It’s not currently possible to add multiple IDs there.

**Reply by adrianjean** [reply]
Ok roger that.
